### PR TITLE
Add new objects & tests

### PR DESCRIPTION
**Pull request by adam-invoiced**
New objects: Coupon, Letter, Note, Task, TaxRate, TextMessage

### CODE REVIEW

**Comment by adam-invoiced** [comment]
> Can you fix what is causing the build to fail?

I'm not sure actually. It appears to be a Travis thing related to Python 3.3. The other two builds are not failing. See

File: invoiced/objects.py [hunk 1]
```diff
@@ -234,13 +324,28 @@ class Plan(CreateableObject, DeleteableObject, ListableObject,
...
     pass
 
-
 class Subscription(CreateableObject, DeleteableObject, ListableObject,
                    UpdateableObject):
 
     def cancel(self):
         return self.delete()
 
+    def preview(self, **params):
```
Comment (jaredtking) [inline invoiced/objects.py]: Also think we should consider doing it this way here: https://github.com/Invoiced/invoiced-python/blob/master/invoiced/objects.py#L71

File: invoiced/util.py [hunk 2]
```diff
@@ -13,6 +13,14 @@ def convert_to_object(_class, values):
     return obj
 
 
+def convert_preview_to_object(_class, values):
```
Comment (jaredtking) [inline invoiced/util.py]: Related to comment about the subscription preview return type

File: invoiced/objects.py (hunk 1 above)
Comment (adam-invoiced) [reply invoiced/objects.py]: This change has been made, as in the other libraries

**Pull request merged**